### DISCUSSION TITLE
Caching a Navigation Menu

### DISCUSSION THREAD

**Discussion by satvikt** [saleor/saleor#10488] [General]
The Navigation menu in question is the **navbar** on our react-storefront homepage, which contains a **nested list of categories** to be displayed.

The problem is that, with the list being quite large, it requires at least 400-500 ms to load on the storefront, leaving the space it occupies empty for the time it takes to load. This causes a bad UX.

My queries:

1. Is there any way to cache a Navigation Menu, like the one mentioned above?
2. Can we use redis for this purpose? If yes, how can redis be configured for this use case?

_(PS: Although Saleor docs and several GitHub discussions mention that redis is used by saleor core for caching, but it seems that there is not enough data available on **What** does redis currently cache and **how** we can use configure redis to cache additional items. Any docs/links on the same will be much appreciated)_

**Comment by slamer59** [comment]
Hello,
I have the same question for many requests and don't  see in the documentation what is cached and not cached. 

Into saleor github code : 
https://github.com/saleor/saleor/blob/7f5db27a203c9985de9b419d6ce9640848eacb20/saleor/graphql/views.py#L11

In Jaeger stack trace no hit to redis database too.

Unfortunately, there is no answers from the team, if you still want this and have time to look, we can try together ?

**Comment by timuric** [comment]
@satvikt ideally this should be cached by your storefront, frameworks like next.js, nuxt or remix provide you with such possibilities. I would not recommend modifying Saleor to add caching, keep in mind that modifying Saleor core in general makes it very difficult to upgrade and alters original behavior, which means it would be difficult to seek support from the community as you would be running essentially a fork.
